presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -152,11 +159,12 @@ protected InternalResourceGroup(Optional<InternalResourceGroup> parent, String n
             id = new ResourceGroupId(name);
             root = this;
         }
+        version = new AtomicInteger();
     }
 
     public ResourceGroupInfo getFullInfo()
     {
-        synchronized (root) {
+        synchronized (getRoot()) {
```

Comment:
What's the reasoning behind figure out what lock to use at runtime vs update `this.root` when it actually changes? Can root group change? Should it change?

Reply:
Also the point of synchronized block is to make sure consistency. If they are actually synchronized on different objects it's not synchronized anymore.

Reply:
it's not about what lock to use but to get to the oldest version of the root via this method. I initially thought to avoid changing the `this.root` vs the lock that is used to synchronized to avoid breaking anything. But it seems `this.root` is not being used anywhere else except to synchronized stuff. So can update the this.root instead.